sensor/common/scannerclient/grpc_client.go

```diff
@@ -164,7 +164,7 @@ func dialV2() (ScannerClient, error) {
 // dialV4 connect to scanner V4 gRPC and return a new ScannerClient.
 func dialV4() (ScannerClient, error) {
 	ctx := context.Background()
-	c, err := client.NewGRPCScanner(ctx, client.WithIndexerAddress(env.ScannerV4IndexerEndpoint.Setting()))
+	c, err := client.NewGRPCScanner(ctx, client.IndexerOnly(), client.WithAddress(env.ScannerV4IndexerEndpoint.Setting()))
```

Comment:
I see you moved away from `WithIndexerAddress()`. Why do we need `IndexerOnly()` now?

Reply:
ah, I see your point now

Reply:
Updated to remove `IndexerOnly`, and I updated the comments for `WithIndexerAddress` and `WithMatcherAddress` to make it clear when to use those